modules/givc/adminvm.nix

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{ config, lib, ... }:
+let
+  cfg = config.ghaf.givc.adminvm;
+  inherit (lib) mkOption mkIf types;
+in
+{
+  options.ghaf.givc.adminvm = {
+    enable = mkOption {
+      description = "Enable adminvm givc module.";
+      type = types.bool;
+      default = true;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Configure admin service
+    givc.admin = {
+      enable = true;
+      inherit (config.ghaf.givc.adminConfig) name;
+      inherit (config.ghaf.givc.adminConfig) addr;
+      inherit (config.ghaf.givc.adminConfig) port;
+      inherit (config.ghaf.givc.adminConfig) protocol;
+      services = [
+        "givc-ghaf-host-debug.service"
+        "givc-net-vm.service"
+        "givc-gui-vm.service"
+        "givc-audio-vm.service"
+      ];
+      tls = {
```

Comment:
This is a minor refactoring proposal of `tls{}`-block. The pattern is repeated in all the VM-declarations so it could be refactored into a module to which the `name ([adminConfig|name|hostName])` is passed. Alternatively, all could use `hostName` so no need to pass the argument.

Reply:
I am gonna remove the TLS related items for now, and add them back in later as a functional solution.